src/index.js

```diff
@@ -117,24 +128,28 @@ function checkArgType (rawText, optionalArgs) {
     .ruled) !== 'undefined') {
     return false
   }
+  if (typeof (optionalArgs.inkColor) !== 'string' && typeof (optionalArgs
```

Comment:
(Optional) This manual input validation is getting very complicated now, Can you rewrite this whole part using [Joi](https://www.npmjs.com/package/joi)?

Reply:
This is optional, I can instead create a new issue for this for someone else to pick up

Reply:
yes, I also thought that keeping track of args is too difficult now, I'll try to incorporate your suggestion.